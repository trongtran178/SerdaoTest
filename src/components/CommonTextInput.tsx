import {StyleSheet, TextInput} from 'react-native';
import React from 'react';
type CommonTextInputProps = {} & React.ComponentProps<typeof TextInput>;

export const CommonTextInput = (props: CommonTextInputProps) => {
  return <TextInput style={styles.textInput} {...props} />;
};

const styles = StyleSheet.create({
  textInput: {
    borderWidth: 1,
    marginVertical: 8,
    width: '80%',
    height: 50,
    borderColor: '#ddd',
    borderRadius: 10,
    paddingHorizontal: 10,
    marginBottom: 15,
    backgroundColor: '#fff',
    elevation: 2, // for Android shadow
    shadowColor: '#000', // for iOS shadow
    shadowOffset: {width: 0, height: 2}, // for iOS shadow
    shadowOpacity: 0.1, // for iOS shadow
    shadowRadius: 5, // for iOS shadow
  },
});
