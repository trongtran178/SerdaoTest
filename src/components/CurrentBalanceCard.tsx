import React from 'react';
import {View, Text, StyleSheet} from 'react-native';

type Props = {
  currentBalance: number;
};

export const CurrentBalanceCard: React.FC<Props> = ({currentBalance}) => {
  return (
    <View style={styles.card}>
      <Text style={styles.balanceText}>Current Balance</Text>
      <Text style={styles.balanceAmount}>{`$${currentBalance.toFixed(
        2,
      )}`}</Text>
    </View>
  );
};

const styles = StyleSheet.create({
  card: {
    padding: 12,
    borderRadius: 12,
    borderWidth: 1,
    borderColor: '#ddd',
    backgroundColor: '#fff',
    elevation: 2, // for Android shadow
    shadowColor: '#000', // for iOS shadow
    shadowOffset: {width: 0, height: 2}, // for iOS shadow
    shadowOpacity: 0.1, // for iOS shadow
    shadowRadius: 5, // for iOS shadow
  },
  balanceText: {
    fontSize: 18,
    fontWeight: 'bold',
    color: '#333',
    marginBottom: 5,
  },
  balanceAmount: {
    fontSize: 24,
    fontWeight: 'bold',
    color: '#4CAF50', // Material Design green
  },
});
