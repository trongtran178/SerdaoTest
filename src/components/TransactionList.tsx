import React, {useCallback} from 'react';
import {
  View,
  Text,
  FlatList,
  StyleSheet,
  Image,
} from 'react-native';
import {Transaction} from '../models';
import {Icons} from '../assets';

type Props = {
  transactions: Transaction[];
};

export const TransactionList: React.FC<Props> = ({transactions}) => {
  const renderListEmptyComponent = useCallback(() => {
    return (
      <View testID='tranhis-list-empty-component' style={styles.emptyListContainer}>
        <Image source={Icons.empty_transaction} style={styles.emptyListIcon} />
        <Text style={styles.emptyListTitle}>No Transaction History</Text>
        <Text style={styles.emptyListMessage}>
          You haven't made any transactions yet. Start by sending or receiving
          money!
        </Text>
      </View>
    );
  }, []);

  const renderItem = ({item}: {item: Transaction}) => (
    <View testID={`transaction-item`} style={styles.transactionItem}>
      <Text testID='transaction-id' style={styles.transactionId}>ID: {item.id}</Text>
      <Text testID='transaction-amount' style={styles.transactionAmount}>
        Amount: ${item.amount.toFixed(2)}
      </Text>
      <Text testID='transaction-to' style={styles.transactionTo}>To: {item.account.name}</Text>
      <Text testID='transaction-iban' style={styles.transactionIban}>IBAN: {item.account.iban}</Text>
    </View>
  );

  return (
    <FlatList
      ListEmptyComponent={renderListEmptyComponent}
      scrollEnabled={false}
      data={transactions}
      renderItem={renderItem}
      keyExtractor={item => item.id}
    />
  );
};

const styles = StyleSheet.create({
  listContainer: {},
  transactionItem: {
    padding: 12,
    marginVertical: 4,
    borderRadius: 10,
    borderWidth: 1,
    borderColor: '#ddd',
    backgroundColor: '#fff',
    elevation: 2, // for Android shadow
    shadowColor: '#000', // for iOS shadow
    shadowOffset: {width: 0, height: 2}, // for iOS shadow
    shadowOpacity: 0.1, // for iOS shadow
    shadowRadius: 5, // for iOS shadow
  },
  transactionId: {
    fontSize: 14,
    color: '#666',
    marginBottom: 5,
  },
  transactionAmount: {
    fontSize: 16,
    fontWeight: 'bold',
    color: '#4CAF50', // Material Design green
    marginBottom: 5,
  },
  transactionTo: {
    fontSize: 14,
    color: '#333',
    marginBottom: 5,
  },
  transactionIban: {
    fontSize: 14,
    color: '#333',
  },
  emptyListContainer: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    padding: 20,
  },
  emptyListIcon: {
    width: 100,
    height: 100,
    marginBottom: 20,
  },
  emptyListTitle: {
    fontSize: 18,
    fontWeight: 'bold',
    color: '#333',
    marginBottom: 10,
  },
  emptyListMessage: {
    fontSize: 16,
    color: '#666',
    marginBottom: 20,
    textAlign: 'center',
  },
  emptyListButton: {
    backgroundColor: '#4CAF50', // Material Design green
    paddingVertical: 15,
    paddingHorizontal: 30,
    borderRadius: 10,
  },
  emptyListButtonText: {
    color: '#fff',
    fontSize: 16,
    fontWeight: 'bold',
  },
});
