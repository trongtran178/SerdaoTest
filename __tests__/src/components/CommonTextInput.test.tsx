import { describe, it, expect } from "@jest/globals"
import { render } from "@testing-library/react-native"
import { TextInput } from "react-native"
import { CommonTextInput } from "../../../src/components/CommonTextInput"

describe('CommonTextInput', () => {

  it('should render correctly', () => {
    const {toJSON} = render(<CommonTextInput />);
    expect(toJSON()).toMatchSnapshot();
    });

  it('should render input with correct text', () => {
    const { getByTestId } = render(
      <CommonTextInput placeholder="Test" testID="test-input" />,
    )
    const input = getByTestId('test-input') as unknown as TextInput
    expect(input.props.placeholder).toBe('Test')
  })
})
