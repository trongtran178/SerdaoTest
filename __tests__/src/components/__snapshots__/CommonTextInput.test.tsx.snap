// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommonTextInput should render correctly 1`] = `
<TextInput
  style={
    {
      "backgroundColor": "#fff",
      "borderColor": "#ddd",
      "borderRadius": 10,
      "borderWidth": 1,
      "elevation": 2,
      "height": 50,
      "marginBottom": 15,
      "marginVertical": 8,
      "paddingHorizontal": 10,
      "shadowColor": "#000",
      "shadowOffset": {
        "height": 2,
        "width": 0,
      },
      "shadowOpacity": 0.1,
      "shadowRadius": 5,
      "width": "80%",
    }
  }
/>
`;
